AutoFinance.Broker/InteractiveBrokers/Controllers/TwsControllerBase.cs

```diff
@@ -696,58 +834,148 @@ public Task<List<PositionStatusEventArgs>> RequestPositions(CancellationToken ca
         {
             // Set the operation to cancel after 10 seconds
             CancellationTokenSource tokenSource = new CancellationTokenSource(10000);
-            return this.RequestSecurityDefinitionOptionParameters(underlyingSymbol, exchange, underlyingSecType, underlyingConId, tokenSource.Token);
+            return RequestSecurityDefinitionOptionParameters(
+                underlyingSymbol,
+                exchange,
+                underlyingSecType,
+                underlyingConId,
+                this.twsRequestIdGenerator,
+                this.twsCallbackHandler,
+                this.clientSocket,
+                tokenSource.Token);
         }
 
         /// <summary>
-        /// Request security definition parameters.
-        /// This is mainly used for finding strikes, multipliers, exchanges, and expirations for options contracts.
+        /// Set the type for the market data feed
         /// </summary>
-        /// <param name="underlyingSymbol">The underlying symbol</param>
-        /// <param name="exchange">The exchange</param>
-        /// <param name="underlyingSecType">The underlying security type</param>
-        /// <param name="underlyingConId">The underlying contract ID, retrieved from the Contract Details call</param>
+        /// <param name="marketDataTypeId">The feed level</param>
+        public void RequestMarketDataType(int marketDataTypeId)
+        {
+            this.clientSocket.RequestMarketDataType(marketDataTypeId);
```

Comment:
Calling this method will produce a callback on the `MarketDataTypeEvent` of the `TwsCallbackHandler`.

Following the pattern, this method should be an async method which listens for the event before returning. `GetAccountDetailsAsync` is a good template method to follow.

Reply:
Ok.

Reply:
It seems that `MarketDataTypeEvent` is triggered by `reqMktData` and not `reqMarketDataType`.
This event has for second argument `int reqId` which corresponds to `int tickerId` from `reqMktData`. 
As seen in https://interactivebrokers.github.io/tws-api/interfaceIBApi_1_1EWrapper.html#a79918660257f0e7ce77309bf3bccf13c.

Reply:
I confirm my previous comment as you will see in the integration tests.